radixdlt-java/src/main/java/org/radix/serialization2/mapper/RadixCBORFactory.java

```diff
@@ -0,0 +1,451 @@
+     */
+
+    @Override
+    protected IOContext _createContext(Object srcRef, boolean resourceManaged) {
+        return super._createContext(srcRef, resourceManaged);
+    }
+
+    /**
+     * Overridable factory method that actually instantiates desired
+     * parser.
+     */
+    @Override
+    protected CBORParser _createParser(InputStream in, IOContext ctxt) throws IOException
+    {
+        return new CBORParserBootstrapper(ctxt, in).constructParser(_factoryFeatures,
+                _parserFeatures, _formatParserFeatures,
+                _objectCodec, _byteSymbolCanonicalizer);
+    }
+
+    /**
+     * Overridable factory method that actually instantiates desired
+     * parser.
+     */
+    @Override
+    protected JsonParser _createParser(Reader r, IOContext ctxt) throws IOException {
+        return _nonByteSource();
+    }
+
+    @Override
+    protected JsonParser _createParser(char[] data, int offset, int len, IOContext ctxt,
+            boolean recyclable) throws IOException {
+        return _nonByteSource();
+    }
+
+    /**
+     * Overridable factory method that actually instantiates desired
+     * parser.
+     */
+    @Override
+    protected CBORParser _createParser(byte[] data, int offset, int len, IOContext ctxt) throws IOException
+    {
+        return new CBORParserBootstrapper(ctxt, data, offset, len).constructParser(
+                _factoryFeatures, _parserFeatures, _formatParserFeatures,
+                _objectCodec, _byteSymbolCanonicalizer);
+    }
+
+    @Override
+    protected RadixCBORGenerator _createGenerator(Writer out, IOContext ctxt) throws IOException {
+        return _nonByteTarget();
+    }
+
+    @Override
+    protected RadixCBORGenerator _createUTF8Generator(OutputStream out, IOContext ctxt) throws IOException {
+        return _createCBORGenerator(ctxt,
+                _generatorFeatures, _formatGeneratorFeatures, _objectCodec, out);
+    }
+
+    @Override
+    protected Writer _createWriter(OutputStream out, JsonEncoding enc, IOContext ctxt) throws IOException {
+        return _nonByteTarget();
+    }
+
+    private final RadixCBORGenerator _createCBORGenerator(IOContext ctxt,
+            int stdFeat, int formatFeat, ObjectCodec codec, OutputStream out) throws IOException
+    {
+        // false -> we won't manage the stream unless explicitly directed to
+    	RadixCBORGenerator gen = new RadixCBORGenerator(ctxt, stdFeat, formatFeat, _objectCodec, out);
+        if (RadixCBORGenerator.Feature.WRITE_TYPE_HEADER.enabledIn(formatFeat)) {
+            gen.writeTag(CBORConstants.TAG_ID_SELF_DESCRIBE);
+        }
+        return gen;
+    }
+
+    protected <T> T _nonByteSource() {
+        throw new UnsupportedOperationException("Can not create parser for non-byte-based source");
+    }
+
+    protected <T> T _nonByteTarget() {
+        throw new UnsupportedOperationException("Can not create generator for non-byte-based target");
+    }
+}
```

Comment:
I understand this is a modified version of some code from another project and that our modifications have been minimal. But as a future note, especially if we need to modify a bit more, it could be nice to commit the original file, in its original state, as a separate commit and then apply our changes in a new commit.